nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -746,6 +769,26 @@ in
           }
         ]));
 
+    warnings =
+      optional (
+        cfg.writableStore &&
+        cfg.useNixStoreImage &&
+        opt.writableStore.highestPrio > lib.modules.defaultPriority)
+        ''
+          You have enabled ${opt.useNixStoreImage} = true,
+          without setting ${opt.writableStore} = false.
+
+          This causes a store image to be written to the store, which is
+          costly, especially for the binary cache, and because of the need
+          for more frequent garbage collection.
+
+          If you really need this combination, you can set ${opt.writableStore}
+          explicitly to false, incur the cost and make this warning go away.
```

Comment:
I'm confused about the message. Shouldn't it say "explicitly to *true*"? This paragraph essentially says "set it to false to incur the cost, otherwise we recommend to set it to false", no?

Reply:
Good catch! https://github.com/NixOS/nixpkgs/pull/186626